### PR TITLE
Refactor AgentBatchTask.java

### PR DESCRIPTION
**Pull request by sduzh**
Reduce time complexity and avoid temporary object creation.

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/task/AgentBatchTask.java [hunk 1]
```diff
@@ -121,11 +116,12 @@ public boolean isFinished() {
...
-        for (List<AgentTask> tasks : this.backendIdToTasks.values()) {
-            for (AgentTask agentTask : tasks) {
+        for (Map.Entry<Long, List<AgentTask>> entry : this.backendIdToTasks.entrySet()) {
+            for (AgentTask agentTask : entry.getValue()) {
                 if (!agentTask.isFinished()) {
-                    if (res.size() < limit) {
-                        res.add(agentTask);
+                    res.add(agentTask);
+                    if (res.size() >= limit) {
+                        return res;
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/task/AgentBatchTask.java:124]: This modification changes the original behaviour.
Whereas the previous behaviour would collect all matches, this modification returns the first non-conforming one it encounters.

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/task/AgentBatchTask.java:124]: I don't quite understand,  IMHO, both implementations will push the first `limit` unfinished tasks into `res`, the difference is that the previous one will iterate all elements, even if the size of `res` has reached to `limit`.

Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/task/AgentBatchTask.java:127]: Oh, you're right.
I misunderstand it.

**Review by imay — APPROVED** [review]

**Comment by chaoyli** [comment]
What's the purpose of this refactor? Could you give some message?

**Comment by sduzh** [comment]
> What's the purpose of this refactor? Could you give some message?

Done, PTAL

**Comment by DeepThinker666** [comment]
could you give the performance boost data in your test?

**Review by chaoyli — APPROVED** [review]

**Pull request merged**